src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/close/AttemptCloseStep.kt

```diff
@@ -41,21 +42,27 @@ class AttemptCloseStep(
 
     @Suppress("TooGenericExceptionCaught")
     override suspend fun execute() {
+        val index = managedIndexMetaData.index
         try {
-            logger.info("Executing close on ${managedIndexMetaData.index}")
+            logger.info("Executing close on $index")
             val closeIndexRequest = CloseIndexRequest()
-                .indices(managedIndexMetaData.index)
+                .indices(index)
 
             val response: CloseIndexResponse = client.admin().indices().suspendUntil { close(closeIndexRequest, it) }
+            logger.info("Close index for $index was acknowledged=${response.isAcknowledged}")
             if (response.isAcknowledged) {
                 stepStatus = StepStatus.COMPLETED
                 info = mapOf("message" to "Successfully closed index")
             } else {
                 stepStatus = StepStatus.FAILED
-                info = mapOf("message" to "Failed to close index: ${managedIndexMetaData.index}")
+                info = mapOf("message" to "Failed to close index")
             }
+        } catch (e: SnapshotInProgressException) {
+            logger.warn("Failed to close index [index=$index] with snapshot in progress")
+            stepStatus = StepStatus.CONDITION_NOT_MET
+            info = mapOf("message" to "Index had snapshot in progress, retrying closing")
         } catch (e: Exception) {
-            logger.error("Failed to set index to close [index=${managedIndexMetaData.index}]", e)
+            logger.error("Failed to set index to close [index=$index]", e)
```

Comment:
we can remove `[index=$index]` here also.

Reply:
Hmm why? We won't know what index it was for then.

Reply:
I thought this was a message for the metadata. You are right we do want to write it in error log.